### PR TITLE
Shorten package names and stop using "_"

### PR DESCRIPTION
**Pull request by mschoch**
Go convention is to not use "_" in package names.  Over time as I've started using more and more static analysis tools, many of them complain about this.  It is arbitrary, but helpful to conform to the conventions.

While doing this, several other names have been shorted.

Much of these changes are internal to Bleve and thus won't affect users.  But, they DO affect any user who has built custom analyzers, as MANY of these packages changed name or location.

Complete list of package name changes in bleve:

```
analysis/analyzers/custom_analyzer => analysis/analyzers/custom
analysis/analyzers/keyword_analyzer => analysis/analyzers/keyword
analysis/analyzers/simple_analyzer => analysis/analyzers/simple
analysis/analyzers/standard_analyzer => analysis/analyzers/standard

analysis/char_filters/html_char_filter => analysis/char/html
analysis/char_filters/regexp_char_filter => analysis/char/regexp
analysis/char_filters/zero_width_non_joiner => analysis/char/zerowidthnonjoiner

datetime_parsers/flexible_go => datetime/flexible
datetime_parsers/datetime_optional => datetime/optional

analysis/language/ar => analysis/lang/ar
analysis/language/bg => analysis/lang/bg
analysis/language/ca => analysis/lang/ca
analysis/language/cjk => analysis/lang/cjk
analysis/language/ckb => analysis/lang/ckb
analysis/language/cs => analysis/lang/cs
analysis/language/el => analysis/lang/el
analysis/language/en => analysis/lang/en
analysis/language/eu => analysis/lang/eu
analysis/language/fa => analysis/lang/fa
analysis/language/fr => analysis/lang/fr
analysis/language/ga => analysis/lang/ga
analysis/language/gl => analysis/lang/gl
analysis/language/hi => analysis/lang/hi
analysis/language/hy => analysis/lang/hy
analysis/language/id => analysis/lang/id
analysis/language/in => analysis/lang/in
analysis/language/it => analysis/lang/it
analysis/language/pt => analysis/lang/pt

analysis/tokenizers/regexp_tokenizer => analysis/tokenizers/regexp
analysis/tokenizers/single_token => analysis/tokenizers/single
analysis/tokenizers/whitespace_tokenizer => analysis/tokenizers/whitespace

analysis/token_map => analysis/tokenmap

analysis/token_filters/apostrophe_filter => analysis/tokens/apostrophe
analysis/token_filters/camelcase_filter => analysis/tokens/camelcase
analysis/token_filters/compound => analysis/tokens/compound
analysis/token_filters/edge_ngram_filter => analysis/tokens/edgengram
analysis/token_filters/elision_filter => analysis/tokens/elision
analysis/token_filters/keyword_marker_filter => analysis/tokens/keyword
analysis/token_filters/length_filter => analysis/tokens/length
analysis/token_filters/lower_case_filter => analysis/tokens/lowercase
analysis/token_filters/ngram_filter => analysis/tokens/ngram
analysis/token_filters/porter => analysis/tokens/porter
analysis/token_filters/shingle => analysis/tokens/shingle
analysis/token_filters/stop_tokens_filter => analysis/tokens/stop
analysis/token_filters/truncate_token_filter => analysis/tokens/truncate
analysis/token_filters/unicode_normalize => analysis/tokens/unicodenorm

index/upside_down => index/upsidedown

numeric_util => numeric

search/highlight/fragment_formatters/ansi => search/highlight/format
search/highlight/fragment_formatters/html => search/highlight/format/html
```

One related change in blevex:

```
detect_lang => detectlang
```

### CODE REVIEW

**Comment by mschoch** [comment]
Review @steveyen

File: test/integration_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/blevesearch/bleve"
 
 	// we must explicitly include any functionality we plan on testing
-	_ "github.com/blevesearch/bleve/analysis/analyzers/keyword_analyzer"
+	//_ "github.com/blevesearch/bleve/analysis/analyzers/keyword_analyzer"
```
Comment (steveyen) [inline test/integration_test.go]: quickie double-check -- wondering if this was commented out line was on purpose?

Comment (steveyen) [inline test/integration_test.go]: if it's on purpose, then all looks good to me

**Comment by mschoch** [comment]
Good catch, ya i saw it and the comment looked odd.  It seemed unnecessary to me based on the other imports so I wanted to try without, and... it passed, so I should remove it.

**Comment by mschoch** [comment]
One package rename was missed and updated with a subsequent commit to bleve:

```
stemmer_filter => stemmer
```

**Pull request merged**